Corona-Warn-App/src/test/java/de/rki/coronawarnapp/service/submission/ScanResultTest.kt

```diff
@@ -27,6 +29,29 @@ class ScanResultTest {
         scanResult = QRScanResult("https://localhost/?$guid")
         scanResult.isValid shouldBe true
 
+        //UPPERCASE and MIXED Cases Strings
+        scanResult = QRScanResult("https://localhost/?$lowerCaseGuid")
+        scanResult.isValid shouldBe true
+
+        scanResult = QRScanResult("HTTPS://LOCALHOST/?$lowerCaseGuid")
+        scanResult.isValid shouldBe true
+
+        scanResult = QRScanResult("https://localhost/?$mixedCaseGuid")
+        scanResult.isValid shouldBe true
+
+        scanResult = QRScanResult("https://LOCALHOST/?$mixedCaseGuid")
+        scanResult.isValid shouldBe true
+
+        //extra slashes should be invalid.
+        scanResult = QRScanResult("HTTPS:///LOCALHOST/?$guid")
+        scanResult.isValid shouldBe false
+
+        scanResult = QRScanResult("HTTPS://LOCALHOST//?$guid")
+        scanResult.isValid shouldBe false
+
+        scanResult = QRScanResult("HTTPS://LOCALHOST///?$guid")
+        scanResult.isValid shouldBe false
+
         // more invalid tests checks
         scanResult = QRScanResult("http://localhost/?$guid")
         scanResult.isValid shouldBe false
```

Comment:
if you want, you could put every test into its own method - this way the result of every test could be seen quickly in circle ci / GitHub

Reply:
@chris-cwa  that's been doen. tried to segregate valid and invalid tests, along with upper lower and mixed case letters